### PR TITLE
✨ Dynamic Endless pagination for trustlist

### PR DESCRIPTION
**Pull request by rayanfer32**
## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor/Revamp ( rewrite or restructure code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I ran `npm run format`/`yarn format` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [ ] I have updated `develop` and merged to my branch before submitting pull request
- [ ] My pull request generate no conflicts with `develop` or `main` branch
- [ ] I requested code review from other team members

## Related Issue

- none

## Proposed Changes

- none

## Additional Info

- none

## Screenshots

|              Original              |              Updated              |
| :--------------------------------: | :-------------------------------: |
|  |  |
|  |  |

### CODE REVIEW

**Review by shrivatsabhat — APPROVED** [review]
looks good for me

**Comment by shrivatsabhat** [comment]
@rayanfer32 please add the total pages number it will affect user experience 
image

user will not know how many pages will be present in this case >>>

File: src/components/Table/DynamicPagination.jsx [hunk 1]
```diff
@@ -72,8 +72,9 @@ export default function DynamicPagination({ controls }) {
             <BiChevronLeft color="inherit" />
           </button>
           <span className={styles.pagination__btn__page}>
-            Page <strong>
-              {pageIndex + 1} of {pageCount}
+            Page{' '}
+            <strong>
+              {pageIndex + 1} {pageCount != Infinity && `of ${pageCount}`}
```
Comment (shrivatsabhat) [inline src/components/Table/DynamicPagination.jsx:77]: this is may not be working as intended

**Review by shrivatsabhat — DISMISSED** [review]

**Comment by rayanfer32** [comment]
This will have a bad experience on switching to the next page,  as for the next page it has to make a network request.

File: src/components/Table/DynamicPagination.jsx (hunk 1 above)
Comment (rayanfer32) [reply src/components/Table/DynamicPagination.jsx:77]: Its working as expected

**Pull request merged**